app/controllers/admin/dashboard_controller.rb

```diff
@@ -0,0 +1,226 @@
+    params[:active] = detect_active params[:active]
+    @users.update_attribute :active, params[:active]
+    respond_to do |format|
+      format.js
+    end
+  end
+
+  # Update Category
+  def update_category
+    @category_data = Category.all.sort_by_newest.page(params[:page])
+    @category_data = @category_data.per_page Settings.admin.category_per_page
+    @category = Category.find_by id: params[:id]
+    if @category.update_attributes category_params
+      respond_to do |format|
+        format.js{render "sucess_update_category.js.erb"}
+      end
+    else
+      respond_to do |format|
+        format.js
+      end
+    end
+  end
+
+  # Update Category Active
+  def active_update_category
+    @category = Category.find_by id: params[:id].to_i
+    params[:active] = detect_active params[:active]
+    @category.update_attribute :active, params[:active]
+    respond_to do |format|
+      format.js
+    end
+  end
+
+  # Update Product
+  def update_product
+    if @update_status
+      respond_to do |format|
+        format.js{render "sucess_update_product.js.erb"}
+      end
+    else
+      respond_to do |format|
+        format.js
+      end
+    end
+  end
+
+  # Update Products Active
+  def active_update_products
+    @products = Product.find_by id: params[:id].to_i
+    params[:active] = detect_active params[:active]
+    @products.update_attribute :active, params[:active]
+    respond_to do |format|
+      format.js
+    end
+  end
+
+  # Destroy Category Product
+  def category_product_destroy
+    session[:category_product].delete(params[:category].to_i)
+    respond_to do |format|
+      format.js{render "category_product_append.js.erb"}
+    end
+  end
+
+  # Destroy Product
+  def product_destroy
+    @product = Product.find_by id: params[:id]
+    ActiveRecord::Base.transaction do
+      begin
+        product_destroy_processing
+      rescue StandardError
+        flash[:warning] = "Can not Delete this data"
+        respond_to do |format|
+          product_location = "window.location.replace('/admin/product_data');"
+          format.js{render js: product_location}
+        end
+      end
+      flash[:success] = "Destroy Product Success"
+      respond_to do |format|
+        format.js{render js: "window.location.replace('/admin/product_data');"}
+      end
+    end
+  end
+
+  private
+
+  def product_params
+    params.require(:product)
+  end
+
+  def category_params
+    params.require(:category).permit :name, :status, :type_food
+  end
+
+  def init_product
+    @product_data = Product.all.sort_by_newest.page(params[:page])
+    @product_data = @product_data.per_page Settings.admin.product_per_page
+    @product = Product.find_by id: params[:id]
+  end
+
+  def load_sort_category
+    type_food = params[:type_food]
+    status = params[:status]
+    @category = sort_by_type type_food
+    @category = sort_by_status status, @category
+  end
+
+  def detect_active param
+    return true if param == "true"
+    false
```

Comment:
toán tử ngôi số 3
`return (param == "true") ? true : false`

mà bỏ xem thử bỏ `return` luôn dc ko nhé, chị nhớ là dc ấy